Quick-Start/Prerequisite.md

```diff
@@ -1,101 +1,101 @@
+ 
+- **Version Compatibility:** Extensions are designed to be compatible across different versions of PostgreSQL. They abstract the implementation details and provide a consistent interface, allowing extensions to work seamlessly across compatible versions. This makes it easier to upgrade or migrate your database while retaining the functionality provided by the extensions. 
+ 
+- **Community Contributions:** PostgreSQL has a vibrant and active community that develops and maintains a wide range of extensions. These extensions cover various domains such as GIS, full-text search, JSON processing, time-series data, and more. By leveraging existing extensions, developers can benefit from community expertise and avoid reinventing the wheel. 
+ 
+- **Performance Optimization:** Extensions can also be used to optimize the performance of your PostgreSQL database. They can provide advanced indexing techniques, caching mechanisms, or query optimization algorithms that improve query execution speed and overall database performance.  
+ 
+Overall, PostgreSQL extensions enhance the flexibility, customization, and extensibility of the PostgreSQL database system. They allow developers and users to tailor the database to their specific needs, leverage community-contributed functionality, and simplify the installation and management of additional features.  
+# Basic Architecture:  
+The extension architecture follows a modular approach, where each extension is packaged as a separate unit. Let's explore the key components and how they work together: 
+ 
+1. **Extension Control File:** 
+    - The extension control file, typically named `extension_name.control`, is a plain text file that provides metadata and instructions about the extension. 
+    - It contains information such as the extension name, version, description, dependencies, installation script, and configuration options. 
+    - This file is used by PostgreSQL to manage the extension's installation, upgrade, and removal. 
+
+2. **Extension SQL File:** 
+    - The extension SQL file, typically named `extension_name.sql`, contains the SQL statements required to create the extension's objects in the database. 
+    - It can include table definitions, function definitions, views, triggers, and other database objects specific to the extension. 
+    - The SQL file is executed during the installation process to set up the extension's functionality in the database. 
+ 
+3. **Extension Library:** 
+    - Some extensions may require additional libraries or shared objects to provide their functionality. 
+    - The extension library is a compiled binary or a collection of source files that implement the extension's core functionality. 
```

Comment:
Well no, the library is compiled from a collection of one or more source files, which are generally written in C.